### ISSUE TITLE
[Feature Request] Edit Gifs

### ISSUE THREAD

**Issue by Vastrix** [ShareX/ShareX#3393]
Hey,  
  
Wouldn't it be cool if we could edit gifs as well?  
Nothing fancy: Remove frame, swap frames, edit frame as picture, etc :)  
  
Regardless, Great Program!

**Comment by Jaex** [comment]
Duplicate of #1554

**Comment by vatterspun** [comment]
I agree with Jaex that it's probably out of scope.  Recommend ScreenToGif here on Github: https://github.com/NickeManarin/ScreenToGif/

**Comment by Vastrix** [comment]
weird, I actually did check for dupes before posting.. :s  
I disagree with it being out of scope tho, You'd like to edit stuff before you share them and you can already edit normal pics so...  
But yea, too much work I spose.. :/  
  
Thanks tho :)

**Comment by vatterspun** [comment]
> you can already edit normal pics

Editing single graphics and editing a series of them is a bit of a leap.  For example, there's edits to multiple frames but not all of them, frame management, cropping, rate of the GIF, color levels, etc. that really takes a lot of additional thought.  Still, Jaex has added a lot of functionality that I thought was out of scope for a screencapture program so maybe he'll figure something out in the future.

**Comment by Vastrix** [comment]
Yea, there are indeed a lot of things you can edit with gifs but but I did suggest to only cover the basics :)  
But yea, it'll probably still be a hassle.. :s

**Comment by stephen147** [comment]
ScreenToGif is even much better than photoshop for gifs!

Thanks for the recommendation @vatterspun.

**Comment by PySimpleGUI** [comment]
Today I gave ShareX a try at the suggestion of a friend.  For years I've used Snagit.  I like their video editor so I gave it a shot on the ShareX GIF.  It didn't work.  

When using Snagit, multiple megabytes are not unusual for a captured GIF.  I was _**stunned**_ to see the size of the GIFs made by ShareX.  When I tried to edit the GIF I was unable to with the Snagit editor.  It seems like the reason has to do with the compression technique used.  I think they're storing entire frames where ShareX is storing the changes only from one frame to another.  I don't know the animated GIF format, but mentally I equate it to B-frame in MPEG versus I-frames where the entire frame is stored.  

I can live with the lack of an editor when the GIF size is so tiny by comparison.
